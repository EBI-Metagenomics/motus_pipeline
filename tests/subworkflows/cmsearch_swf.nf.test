nextflow_process {

    name "Test SubWorkflow cmsearch-easel"
    script "subworkflows/cmsearch_swf.nf"
    process "CMSEARCH_SUBWF"

    test("cmsearch-deoverlap-extract") {

        when {
            params {
                seq = "${baseDir}/tests/modules/fixtures/cmsearch/test.fasta"
                cov = "${baseDir}/tests/modules/fixtures/cmsearch/"
                clan = "${baseDir}/tests/modules/fixtures/cmsearch/ribo.claninfo"
            }
            process {
                """
                input[0] = channel.value("mgtest")
                input[1] = channel.fromPath("${params.seq}", checkIfExists: true)
                input[2] = channel.fromPath("${params.cov}/*.cm", checkIfExists: true)
                input[3] = channel.fromPath("${params.clan}", checkIfExists: true)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
}