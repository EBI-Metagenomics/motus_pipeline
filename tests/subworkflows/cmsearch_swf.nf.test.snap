{
    "cmsearch-deoverlap-extract": {
        "content": [
            {
                "cmsearch_lsu_fasta": [
                    
                ],
                "cmsearch_ssu_fasta": [
                    
                ],
                "seq_cat": [
                  [
                      [
                          "mgtest_RNaseP_bact_b.fasta:md5,7eefda3f2ae374918ddeb201933bca66"
                      ]
                  ]
                ]
            }
        ],
        "timestamp": "2023-03-23T09:56:38+0000"
    }
}